### PR TITLE
fix: [OPA-1039]: Adding On Prem check for OPA

### PR DESCRIPTION
**Pull request by kylekstewart**
### Summary

We need to add a check for all OPA links/areas to show when on prem, as it was only showing based off of licenses.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by knagurski — APPROVED** [review]
👍🏻

**Review by bgribben-harness — APPROVED** [review]

File: src/modules/10-common/hooks/useModuleLicenses.tsx [hunk 1]
```diff
@@ -6,12 +6,15 @@
...
 
 import { Editions } from '@common/constants/SubscriptionTypes'
+import { isOnPrem } from '@common/utils/utils'
 import { useLicenseStore } from 'framework/LicenseStore/LicenseStoreContext'
 import { useModuleInfo } from './useModuleInfo'
 
 export function useAnyEnterpriseLicense(): boolean {
   const { licenseInformation } = useLicenseStore()
 
+  if (isOnPrem()) return true
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/hooks/useModuleLicenses.tsx:16]: have we considering all the usecases of `useAnyEnterpriseLicense`? returning true without license check for all onprem customers seems risky.

**Review by knagurski — APPROVED** [review]
Still looks good. But worth clarifying the answer to @abhinavrastogi-harness's question

File: src/modules/10-common/hooks/useModuleLicenses.tsx (hunk 1 above)
Comment (arya-harness) [reply src/modules/10-common/hooks/useModuleLicenses.tsx:16]: on onprem, licensing is scheduled for next release. This bypass was needed for pov customer(CBP). We are planning to remove all such cases once licensing has been enforced.
TIA.

Comment (abhinavrastogi-harness) [reply src/modules/10-common/hooks/useModuleLicenses.tsx:16]: Question is answered here: https://harness.slack.com/archives/C04EB0E2YES/p1670585783752269

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**